### PR TITLE
Add Scala 3 support

### LINKED ISSUE #478: Scala 3 Support
Blocked by
- [ ] [zio-schema](https://github.com/zio/zio-schema/issues/50)

### PR DESCRIPTION
**Pull request by mijicd**
Closes #478

### CODE REVIEW

File: project/BuildHelper.scala [hunk 1]
```diff
@@ -36,17 +41,35 @@ object BuildHelper {
...
       scalacOptions += "-language:experimental.macros",
       libraryDependencies ++= {
-        List(
-          "org.scala-lang" % "scala-reflect"  % scalaVersion.value % Provided,
-          "org.scala-lang" % "scala-compiler" % scalaVersion.value % Provided
-        )
+        if (scalaVersion.value == Scala3) Seq()
+        else
+          Seq(
+            "org.scala-lang" % "scala-reflect"  % scalaVersion.value % "provided",
```
Comment (jxnu-liguobin) [inline project/BuildHelper.scala:47]:
```suggestion
            "org.scala-lang" % "scala-reflect"  % scalaVersion.value % Provided,
```

File: build.sbt [hunk 2]
```diff
@@ -33,30 +35,32 @@ lazy val redis =
...
     .enablePlugins(BuildInfoPlugin)
-    .settings(stdSettings("zio-redis"))
     .settings(buildInfoSettings("zio.redis"))
+    .settings(scala3Settings)
+    .settings(stdSettings("zio-redis"))
     .settings(
-      libraryDependencies ++= Seq(
-        "dev.zio"                %% "zio-streams"             % Zio,
+      libraryDependencies ++= List(
+        "dev.zio"                %% "zio-streams"             % "2.0.1",
```
Comment (sideeffffect) [inline build.sbt:43]: Using a variable instead of a literal might be a good idea.

**Review by sideeffffect — APPROVED** [review]
One suggestion, but otherwise looks good to me :+1: 
Thanks!

File: build.sbt (hunk 2 above)
Comment (mijicd) [reply build.sbt:43]: I agree. In fact, I plan to consolidate all of the dependencies that way, but I'll do that in a follow-up.

**Pull request merged**